ocaml/libs/vhd/vhd_format/f.ml

```diff
@@ -2653,11 +2657,16 @@ functor
           return (Cons (x, fun () -> coalesce_request None s))
       | Cons (`Copy (h, ofs, len), next), None ->
           coalesce_request (Some (`Copy (h, ofs, len))) (next ())
-      | Cons (`Copy (h, ofs, len), next), Some (`Copy (h', ofs', len')) ->
+      | Cons (`Copy (h, ofs, len), next), Some (`Copy (h', ofs', len'))
+        when coalesced_sectors ** Int64.of_int sector_size <= sync_limit ->
```

Comment:
I think it's fine - but I stumbled when we have `when` followed by `if`: the if condition could be folded into the `when` and creating a new match for the `else`.